packages/vscode-extension/src/commonlib/azureLogin.ts

```diff
@@ -370,9 +389,16 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
     }
   }
 
-  // TODO add login and select subscription logic later
-  getSelectedSubscription(triggerUI=false): Promise<SubscriptionInfo | undefined> {
+  async getSelectedSubscription(triggerUI = false): Promise<SubscriptionInfo | undefined> {
     const azureAccount = this.getAzureAccount();
+    if (triggerUI) {
+      if (azureAccount.status != loggedIn) {
+        await this.login(true);
+      }
+      if (azureAccount.status === loggedIn && !AzureAccountManager.subscriptionId) {
+        this.selectSubscription();
+      }
+    }
     if (azureAccount.status === loggedIn) {
```

Comment:
Is it possible that account is logged in, but no subs selected?

Reply:
Yes, login through treeView and have not selected subscription.

Reply:
If so, return undefined (not empty `SubscriptionInfo`) since no subs is selected.

Reply:
For this method, if a user logins in and cancels select subscription, we will throw an error.

Reply:
Changed to this:
```
if login and select sub:
    return SubscriptionInfo;
else if login and only have one sub:
    return SubscriptionInfo;
else 
    return undefined;
```

```
if trigger === true and (cancel login or select sub):
    throw error;
else
    return SubscriptionInfo;
```